cmd/tnf/addclaim/addclaim_test.go

```diff
@@ -47,3 +48,227 @@ func TestNewCommand(t *testing.T) {
+					},
+				},
+				testcaseResult2: []testCase{
+					{
+						Name:   "[It] observability observability-container-logging [common, observability, observability-container-logging]",
+						Status: "failed",
+					},
+				},
+			},
+			wantDiffresult: []testCase{
+				{
+					Name:   "[It] observability observability-container-logging [common, observability, observability-container-logging]",
+					Status: "skipped",
+				},
+			},
+			wantNotFoundtest:  []string{},
+			wantNotFoundtest2: []string{"[It] observability observability-crd-status [common, observability, observability-crd-status]"},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			gotDiffresult, gotNotFoundtest, gotNotFoundtest2 := compare2TestCaseResults(tt.args.testcaseResult1, tt.args.testcaseResult2)
+			if !reflect.DeepEqual(gotDiffresult, tt.wantDiffresult) {
+				t.Errorf("compare2TestCaseResults() gotDiffresult = %v, want %v", gotDiffresult, tt.wantDiffresult)
+			}
+			if !reflect.DeepEqual(gotNotFoundtest, tt.wantNotFoundtest) {
+				t.Errorf("compare2TestCaseResults() gotNotFoundtest = %v, want %v", gotNotFoundtest, tt.wantNotFoundtest)
+			}
+			if !reflect.DeepEqual(gotNotFoundtest2, tt.wantNotFoundtest2) {
+				t.Errorf("compare2TestCaseResults() gotNotFoundtest2 = %v, want %v", gotNotFoundtest2, tt.wantNotFoundtest2)
+			}
+		})
+	}
+}
+
+//nolint:funlen
+func Test_compare2cnis(t *testing.T) {
+	type args struct {
+		cniList1 cnistruct
+		cniList2 cnistruct
+		nodeName string
+	}
+	tests := []struct {
+		name               string
+		args               args
+		wantDiffplugins    cnistruct
+		wantNotFoundNames  []string
+		wantNotFoundNames2 []string
+	}{
+		{
+			name: "test1",
+			args: args{
+				cniList1: cnistruct{
+					{
+						Name:    "podman",
+						Plugins: nil,
+					},
+					{
+						Name:    "crio",
+						Plugins: nil,
+					},
+				},
+				cniList2: cnistruct{
+					{
+						Name:    "podman",
+						Plugins: nil,
+					},
+				},
+				nodeName: "master1",
+			},
+			wantDiffplugins:    nil,
+			wantNotFoundNames:  []string{"crio"},
+			wantNotFoundNames2: []string{},
+		},
+		{
+			name: "test2",
+			args: args{
+				cniList1: cnistruct{
+					{
+						Name:    "podman",
+						Plugins: nil,
+					},
+				},
+				cniList2: cnistruct{
+					{
+						Name:    "podman",
+						Plugins: nil,
+					},
+					{
+						Name:    "crio",
+						Plugins: nil,
+					},
+				},
+				nodeName: "master1",
+			},
+			wantDiffplugins:    nil,
+			wantNotFoundNames:  []string{},
+			wantNotFoundNames2: []string{"crio"},
+		},
+		{
+			name: "test3",
+			args: args{
+				cniList1: nil,
+				cniList2: cnistruct{
+					{
+						Name:    "podman",
+						Plugins: nil,
+					},
+					{
+						Name:    "crio",
+						Plugins: nil,
+					},
+				},
+				nodeName: "master1",
+			},
+			wantDiffplugins:    nil,
+			wantNotFoundNames:  nil,
+			wantNotFoundNames2: nil,
+		},
+		{
+			name: "test4",
+			args: args{
+				cniList1: nil,
```

Comment:
Maybe it'd be good having another case for empty lists, instead of nil.
```
Case 1:
cniList1: cnistruct{}
cniList2: cnistruct{
			{
				Name:    "podman",
				Plugins: nil,
			},

Case 2:
cniList1: cnistruct{
			{
				Name:    "podman",
				Plugins: nil,
			},
cniList2: cnistruct{}			
```